cri-purge.sh

```diff
@@ -115,45 +131,52 @@ __process_images() {
   echo "Total Images: ${TOTAL_CRI_IMAGES} Unique Images Names: ${TOTAL_UNIQUE_IMAGE_NAMES}"
   echo
   COUNT=0
-  for IMAGE_NAME in ${UNIQUE_CRI_IMAGE_NAMES}; # Do not quote UNIQUE_CRI_IMAGE_NAMES, breaks for loop
+  while read IMAGE_NAME; # Do not quote UNIQUE_CRI_IMAGE_NAMES, breaks for loop
   do
     ((COUNT=COUNT+1))
     echo -n "${COUNT} / ${TOTAL_UNIQUE_IMAGE_NAMES} : Image: ${IMAGE_NAME}"
 
     # Find all versions of this IMAGE_NAME
-    mapfile -t IMAGES <<< "$(echo "${CRI_IMAGES}" | grep "${IMAGE_NAME}")"
+    mapfile -t IMAGES <<< "$(grep "${IMAGE_NAME}" /tmp/CRI_IMAGES)"
     NUM_IMAGES=${#IMAGES[@]}
 
     # If only 1 version detected, keep it.
     if [[ ${NUM_IMAGES} -eq 1 ]]; then
-      echo " - Keep TAG: $(echo "${IMAGES[0]}" | awk '{ printf "%s (%s)\n", $2, $4 }')"
+      TAG_REF=$(__get_tag_ref 0)
+      echo " - Keep TAG: $(echo "${IMAGES[0]}" | awk "{ printf \"%s (%s)\n\", $TAG_REF, \$4 }")"
     else
       # print last line of the array, should be one to keep.
-      echo " - Keep TAG: $(printf %s\\n "${IMAGES[@]: -1}"| awk '{ printf "%s (%s)\n", $2, $4 }')"
+      TAG_REF=$(__get_tag_ref 1)
+      echo " - Keep TAG: $(printf %s\\n "${IMAGES[@]: -1}"| awk "{ printf \"%s (%s)\n\", $TAG_REF, \$4 }")"
 
       for (( i=0; i<$(( ${#IMAGES[@]}-1 )); i++ )) 
       do
         # Remove image if $1 == "PURGE"
+        TAG_REF=$(__get_tag_ref $i)
         if [ "${1^^}" == "PURGE" ]; then
-          echo "- Purge TAG: $( echo "${IMAGES[$i]}" | awk '{ printf "%s (%s)\n", $2, $4 }')"
-
-          # Remove the Specific Image:TAG
-          ${CRI_CMD} rmi "$(echo "${IMAGES[$i]}" |awk '{ printf "%s:%s\n", $1, $2 }')" > /dev/null 2>&1
+          echo "- Purge TAG: $( echo "${IMAGES[$i]}" | awk "{ printf \"%s (%s)\n\", $TAG_REF, \$4 }")"
+
+          # Remove the Specific Image:TAG, or by a hash (if tag=<none>)
```

Comment:
I like being able to handle the "tag=<none>", but how do you determine which hash is the one to keep? I wasn't sure how to deal with this, which is why skipped them.  I guess it could be fine to just keep one hash, purge the others and hope for the best. If the guess is wrong, it would simply be downloaded again anyway... but you still have the benifit of reclaimed disk space.